controller/k8s/endpoints.go

```diff
@@ -129,7 +129,7 @@ type informer struct {
 
 func (i *informer) run() error {
 	go i.informer.Run(i.stopCh)
-	return newWatcher(i.informer, endpointResource).run()
+	return newWatcher(i.informer, endpointResource, nil, i.stopCh).run()
```

Comment:
Ah, it's a bit unfortunate that the method signature for `i.informer.Run` doesn't match the method signature for a `watchInitializer` (it doesn't return `error`). I think we should coerce the `Run` function to match the method signature, and then pass that function as an argument to `newWatcher`. Something like:

```
func (i *informer) run() error {
	initializer := func(stopCh <-chan struct{}) error {
		i.informer.Run(stopCh)
		return nil
	}
	return newWatcher(i.informer, endpointResource, initializer, i.stopCh).run()
}
```

Reply:
I've noticed that even with this fix it's still possible for the process to exit on startup, but fortunately it exits immediately, which is better than hanging for 30 seconds:

```
  destination:
    ...
    State:          Running
      Started:      Mon, 05 Mar 2018 09:16:17 -0800
    Last State:     Terminated
      Reason:       Error
      Exit Code:    2
      Started:      Mon, 05 Mar 2018 09:16:15 -0800
      Finished:     Mon, 05 Mar 2018 09:16:15 -0800
    Ready:          True
    Restart Count:  1
    ...
```

I think it has something to do with our handling of the endpoints watcher that's different from the pods and replicasets watchers. In the long run we should standardize these things, but in the short term I think this is still an ok fix.